### ISSUE TITLE
Improve formatting of multidimensional arrays

### ISSUE THREAD

**Issue by jnyrup** [fluentassertions/fluentassertions#991]
When calling `Formatter.ToString(object)` on a jagged array such as
```c#
new int[][]
{
    new[] { 1, 2 },
    new[] { 3, 4 }
};
```

the output contains the nested structure as expected.
```
{{1, 2}, {3, 4}}
```

Calling `Formatter.ToString(object)` on a multidimensional array such as
```c#
new int[,]
{
    { 1, 2 },
    { 3, 4 }
};
```

gives an unexpected (and perhaps confusing) visual result, as it doesn't contain the nested structure.
```
{1, 2, 3, 4}
```

This could probably be improved by introducing a new `MultidimensionalArrayFormatter` which checks if the `object` is an `Array` and has `Rank >= 2`.

**Comment by Evangelink** [comment]
@jnyrup I am just wondering whether you expect to have a different visual representation for jagged and multidimensional arrays? 

Just in case my question is not really clear, do you expect to also have

```
{{1, 2}, {3, 4}}
```

for multidimensional arrays?

**Comment by jnyrup** [comment]
@Evangelink I would expect jagged and multidimensional arrays to have a *similar* visual representation, i.e. I can't at the top of my head think of a case where I *wouldn't* want them to have the same visual representation.


For multidimensional arrays I think
```
{{1, 2}, {3, 4}}
```
would be an improvement over the current
```
{1, 2, 3, 4}
```

Maybe both jagged and multidimensional arrays could even be printed this?
```
{
  {1, 2},
  {1, 3}
}
```

**Comment by DmitriyMaksimov** [comment]
I implemented the required formater and unit tests.
Before I submit pull request I would like to ask a question - do I need to update documentation? Can't find any related topic.

**Comment by dennisdoomen** [comment]
> do I need to update documentation?

Nah. I don't think we mention the exact output anywhere else.

**Comment by jnyrup** [comment]
Fixed in #1044
